packages/block-editor/src/components/provider/with-registry-provider.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { withRegistry, createRegistry, RegistryProvider } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { storeConfig } from '../../store';
+import applyMiddlewares from '../../store/middlewares';
+
+const withRegistryProvider = createHigherOrderComponent( ( WrappedComponent ) => {
+	return withRegistry( ( { useSubRegistry = true, registry, ...props } ) => {
+		// Disable reason, this rule conflicts with the React hooks rule (no conditionals)
+		// eslint-disable-next-line @wordpress/no-unused-vars-before-return
+		const [ subRegistry, updateRegistry ] = useState( null );
+
+		if ( ! useSubRegistry ) {
+			return <WrappedComponent registry={ registry } { ...props } />;
+		}
+
+		useEffect( () => {
+			const newRegistry = createRegistry( {}, registry );
```

Comment:
Would it work as well?
```suggestion
			const newRegistry = createRegistry( { 'core/block-editor': storeConfig }, registry );
```
This is what I read from:
https://github.com/WordPress/gutenberg/blob/master/packages/data/src/registry.js#L181

Reply:
Yes, previously this was not possible because the controls plugins was not built-in but we can do this now. Good suggestion.

Reply:
Oh I guess we still can't do that yet because we need the returned "store" object to apply middlewares.